cli/pcluster/networking/network_resource_creation.py

```diff
@@ -0,0 +1,235 @@
+    def setup(self, vpc_id, name=None):
+        """
+        Set the parameters necessary for a vpc to be pcluster-compatible.
+
+        :param vpc_id: the target vpc_id
+        :param name: the name that you want to give to the vpc
+        :raise RuntimeError: if some problems occurred during the operation
+        """
+        vpc = self.ec2.Vpc(vpc_id)
+        if name:
+            self.ec2.create_tags(Resources=[vpc_id], Tags=[{"Key": "Name", "Value": name}])
+        vpc.modify_attribute(EnableDnsHostnames={"Value": True})
+        vpc.modify_attribute(EnableDnsSupport={"Value": True})
+
+    @_ExceptionHandler.handle_client_exception
+    def check(self, vpc_id):
+        """
+        Check whether the given vpc respects the condition needed for pcluster.
+
+        :param vpc_id: the target vpc_id
+        :return: True if the vpc is pcluster compatible
+        :raise RuntimeError: if some problems occurred during the operation
+        """
+        vpc = self.ec2.Vpc(vpc_id)
+        dns_resolution = vpc.describe_attribute(Attribute="enableDnsSupport")["EnableDnsSupport"]["Value"]
+        dns_hostnames = vpc.describe_attribute(Attribute="enableDnsHostnames")["EnableDnsHostnames"]["Value"]
+        # default is equal to NO dhcp options set
+        return dns_resolution and dns_hostnames and vpc.dhcp_options_id != "default"
+
+
+class CFNSubnetsGenerator:
+    """Generate a yaml file for cloudformation that handles the creation of subnets inside a vpc."""
+
+    def __init__(self):
+        self.template = Template()
+
+    @staticmethod
+    def _add_name(resource_name):
+        return [{"Key": "Name", "Value": f"pcluster-{resource_name}"}]
+
+    def _add_parameter(self, name, description, expected_input_type):
+        return self.template.add_parameter(Parameter(name, Description=description, Type=expected_input_type))
+
+    def _add_output(self, name, description, value):
+        return self.template.add_output(Output(name, Description=description, Value=value))
+
+    def _write_template(self, path):
+        with open(path, "w+") as file:
+            file.write(self.template.to_yaml())
+
+    def _create_gateway(self, vpc, gateway_attachment_name):
+        internet_gateway = self.template.add_resource(
+            InternetGateway("InternetGateway", Tags=CFNSubnetsGenerator._add_name("Internet-Gateway"))
+        )
+        self.template.add_resource(
+            VPCGatewayAttachment(gateway_attachment_name, VpcId=Ref(vpc), InternetGatewayId=Ref(internet_gateway))
+        )
+        return internet_gateway
+
+    def _create_route_table_with_ig(self, vpc, internet_gateway, gateway_attachment_name, subnet):
+        route_table = self.template.add_resource(
+            RouteTable("RouteTable", VpcId=Ref(vpc), Tags=CFNSubnetsGenerator._add_name("Public-Route-Table"))
+        )
+        self.template.add_resource(
+            Route(
+                "Route",
+                DependsOn=gateway_attachment_name,
```

Comment:
is this needed?

Reply:
Yes, I tested that.

Reply:
Yes it is. Not using that creates the following error:

`
route table rtb-0d007d59751c85f69 and network gateway igw-093c8118a00979282 belong to different networks (Service: AmazonEC2; Status Code: 400; Error Code: InvalidParameterValue; Request ID: e6955710-6e1d-4a9a-a408-d721bfe5274a)
`